### ISSUE TITLE
A note on data augmenation

### ISSUE THREAD

**Issue by taoari** [apache/incubator-mxnet#4447]
In the following figure, I compare commonly used data augmentation techniques for ResNet-50 on ImageNet. **Random crop, random flip, and scale augmentations** were shown to be very **effective**, and they provide as the baseline "scale aug". For **JPEG compression with quality 90** and **aspect ratio augmentation**, the curves indicates they almost **have no effects**. For **color augmentation**, it is shown that the accuracy is actually **decreased**.

![notes_on_aug](https://cloud.githubusercontent.com/assets/3815006/21560667/6f909480-ce9d-11e6-9340-9d8af5f41837.png)

### Compare to state-of-the-art implementation:

 (https://github.com/tornadomeet/ResNet)

For ResNet-50 on ImageNet, top-1 accuracy of 74.55% were achieved. As suggested in https://github.com/tornadomeet/ResNet, canceling scale/color/aspect augmentation will further improve the results. For a fair comparison, https://github.com/tornadomeet/ResNet actually achieved 74.20% accuracy at the 95-th epoch (right before the cancellation of scale/color/aspect augmentations). Hence, we can conclude that **augmentations beyond random crop, random flip, and scale actually hurt the performance** as (74.55%>74.20%).

**Comment by piiswrong** [comment]
I recommend finish training. It's hard to draw conclusions at 40 epochs

**Comment by taoari** [comment]
@piiswrong I have a full training of ResNet-18 on ImageNet. With only Random crop, random flip, and scale augmentations, I am able to achieve an error rate of 30.5%, which is lower than 30.9% in https://github.com/tornadomeet/ResNet. So the conclusion should be valid. Since the evaluation is quite computational expensive. I will continue to train the models when spare GPUs are available.

**Comment by tornadomeet** [comment]
@taoari ResNet-18 which i trained is very bad!, maybe for large batch-size, please ref the implementationof Microsoft which using R in MXNet, which >1% better than mine.

**Comment by taoari** [comment]
@tornadomeet That actually does not matter, both of them were trained on a batch size of 512 for ResNet-18. The point is that **augmentations beyond random crop, random flip, and scale actually hurt the performance**.

In the above figure, I have included the comparison results for scale/aspect ratio/color augmentations. To compare with more advanced techniques, including random rotation, random shear, etc. augmentations, I have included your results.  For a fair comparison, I am using your results at the 95-th epoch. For ResNet-50, only scale augmentation achieved 74.55%, while scale/aspect ratio/color/rotation/shear  augmentations achieved 74.20%. For ResNet-18, since I had a full training (including the cancellation of the augmentations after the 95-th epoch), only scale augmentation achieved an error rate of 30.5%, while scale/aspect ratio/color/rotation/shear  augmentations achieved an error rate of 30.9%. Hence, the conclusion that  **augmentations beyond random crop, random flip, and scale actually hurt the performance** is valid.

For the curve above, we can also conclude that aspect ratio augmentation and JPEG compression almost had no effects on the final results. While color augmentation does hurt the performance.
